### PR TITLE
fix/RSS-ECOMM-4_no-issue: Fix cart flow for user flow

### LINKED ISSUE #186: RSS-ECOMM-4_no-issue: Fix cart flow for user flow
Update cart on user change

### PR DESCRIPTION
**Pull request by Gromov-Roman**
#### Fix cart flow for user flow

resolve #186 

#### 🖊️ Description

Update cart on user change

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [x] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.
- [ ] 👌 Have at least 1 approve.

### CODE REVIEW

File: src/core/api/hooks/useAuth.ts [hunk 1]
```diff
@@ -13,6 +14,7 @@ export const setUserLoading = (isLoading: boolean) => {
 export const setUser = (newUser: Customer | null | undefined) => {
   userSignal.value = newUser;
   setUserLoading(false);
+  loadCart();
```
Comment (JsPowWow) [inline src/core/api/hooks/useAuth.ts:17]: await loadCart()  ???

Comment (Gromov-Roman) [reply src/core/api/hooks/useAuth.ts:17]: it works without await, everything is ok

Comment (Gromov-Roman) [reply src/core/api/hooks/useAuth.ts:17]: works because of signal for cart

Comment (JsPowWow) [reply src/core/api/hooks/useAuth.ts:17]: even so - looks crappy
Can You do await loadCart() after seting user in login, here https://github.com/Maksim99745/eCommerce-Application/pull/187/files#diff-5b91292ed3ca5f7f7c9d71c2d670d22730ab02c8646ab4ff68da962408f44686R26 ?

**Review by JsPowWow — APPROVED** [review]

**Pull request merged**